### PR TITLE
Allow solr filters to be configurable on a per-exhibit basis

### PR DESCRIPTION
**Pull request by jcoyne**
This still needs a controller test, but I wanted your feedback before you head out.

### CODE REVIEW

File: lib/spotlight/engine.rb [hunk 1]
```diff
@@ -87,7 +87,12 @@ def self.blacklight_config
 
     # A lambda expression that filters the solr index per exhibit
     config.exhibit_filter = lambda do |exhibit|
-      { :"#{config.solr_fields.prefix}spotlight_exhibit_slug_#{exhibit.slug}#{config.solr_fields.boolean_suffix}" => true }
+      filter = exhibit.exhibit_filters.first
+      if filter
+        { filter.field => RSolr.solr_escape(filter.value) }
```
Comment (cbeer) [inline lib/spotlight/engine.rb]: Can we push this logic up into `ExhibitFilter`?

File: db/migrate/20151217211019_create_spotlight_exhibit_filters.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateSpotlightExhibitFilters < ActiveRecord::Migration
+  def change
+    create_table :spotlight_exhibit_filters do |t|
+      t.string :field
+      t.string :value
+      t.references :spotlight_exhibit, index: true, foreign_key: true
```
Comment (cbeer) [inline db/migrate/20151217211019_create_spotlight_exhibit_filters.rb]: I think if this was just `references :exhibit`, everything would work out automagically?

File: app/views/spotlight/exhibits/edit.html.erb [hunk 3]
```diff
@@ -6,6 +6,9 @@
       <li role="presentation" class="active">
         <a href="#basic" aria-controls="basic" role="tab" data-toggle="tab"><%= t(:'.basic_settings.heading') %></a>
       </li>
+      <li role="presentation">
```
Comment (cbeer) [inline app/views/spotlight/exhibits/edit.html.erb]: Could we add + check an ability? `can :manage, current_exhibit.filters.build` maybe?

File: app/controllers/spotlight/exhibit_filters_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+module Spotlight
+  # Create and update filters for an exhibit
+  class ExhibitFiltersController < ApplicationController
+    before_action :authenticate_user!
+    load_and_authorize_resource :exhibit, class: 'Spotlight::Exhibit'
+    load_and_authorize_resource through: :exhibit
+
+    def create
+      if @exhibit_filter.save
+        flash[:notice] = 'Filter added'
```
Comment (cbeer) [inline app/controllers/spotlight/exhibit_filters_controller.rb]: I think we have some i18n magic that maybe do this for us?

File: .rubocop.yml [hunk 5]
```diff
@@ -41,5 +41,6 @@ Metrics/ModuleLength:
 
 Metrics/ClassLength:
   Exclude:
+    - 'app/models/spotlight/exhibit.rb'
```
Comment (cbeer) [inline .rubocop.yml]: I'll try to figure out another logical concern to pull out of `Exhibit`..

File: app/views/exhibit_filters/_edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<div role="tabpanel" class="tab-pane" id="filter">
+
+<%= bootstrap_form_for [@exhibit, @exhibit.exhibit_filters.first_or_initialize] do |f| %>
```
Comment (cbeer) [inline app/views/exhibit_filters/_edit.html.erb]: I've asked @ggeisler to give some thought to the UI for this behavior. I don't think it should necessarily hold up the PR, though.

File: app/views/exhibit_filters/_edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  <%= f.text_field :field %>
+  <%= f.text_field :value %>
+  <%= f.submit %>
```
Comment (cbeer) [inline app/views/exhibit_filters/_edit.html.erb]: I think we have some wrapping `div`s for this for other forms?

**Comment by cbeer** [comment]
:+1: Makes sense to me, couple of minor comments, but the idea looks promising.

File: db/migrate/20151217211019_create_spotlight_exhibit_filters.rb (hunk 2 above)
Comment (jcoyne) [inline db/migrate/20151217211019_create_spotlight_exhibit_filters.rb]: Yeah, I was chasing a bug where Postgres wansn't generating the correct foreign_key constraint, but that may only exist when both models are not in the same namespace.  I'll try it the simplest way again.

File: app/views/exhibit_filters/_edit.html.erb (hunk 7 above)
Comment (jcoyne) [inline app/views/exhibit_filters/_edit.html.erb]: It seems like bootstrap_form_for writes out all the correct divs.

File: spec/models/spotlight/exhibit_spec.rb [hunk 8]
```diff
@@ -179,6 +191,9 @@
     before do
       allow(subject).to receive(:blacklight_config).and_return(blacklight_config)
       allow(subject).to receive(:slug).and_return(slug)
+      allow(subject).to receive(:exhibit_filters)
+        .and_return([Spotlight::ExhibitFilter.new(field: subject.send(:default_filter_field),
```
Comment (jcoyne) [inline spec/models/spotlight/exhibit_spec.rb]: Any idea why rubocop isn't catching this style violation?

**Pull request closed without merging**